rules/PSR4/Rector/FileWithoutNamespace/NormalizeNamespaceByPSR4ComposerAutoloadRector.php

```diff
@@ -159,4 +167,41 @@ private function refactorFileWithoutNamespace(
 
         return $namespace;
     }
+
+    /**
+     * @param FileWithoutNamespace|Namespace_ $node
+     * @return array<string,string>
+     */
+    private function getClassLikeRenames(Node $node, string $expectedNamespace): array
+    {
+        $originalNamespace = $node->name?->toString() ?? '';
+        $classLikes = $this->findNonAnonymousClassLikes($node);
+        $classLikesNames = array_map(
+            function (ClassLike $class): string {
+                Assert::notNull($class->name);
+                return $class->name->toString();
+            },
+            $classLikes
+        );
+        $oldClassLikeNames = array_map(fn (string $name): string => $originalNamespace . '\\' . $name, $classLikesNames);
+        $newClassLikeNames = array_map(fn (string $name): string => $expectedNamespace . '\\' . $name, $classLikesNames);
+        $renames = array_combine($oldClassLikeNames, $newClassLikeNames);
+        return $renames;
+    }
+
+    /**
+     * @return ClassLike[]
+     */
+    private function findNonAnonymousClassLikes(Node $node): array
+    {
+        $classLikes = $this->betterNodeFinder->findInstanceOf([$node], ClassLike::class);
+
+        return array_filter($classLikes, static function (ClassLike $classLike): bool {
+            if (! $classLike instanceof Class_) {
+                return true;
+            }
+
+            return ! $classLike->isAnonymous();
```

Comment:
there is `ClassAnalyzer->isAnonymousClass()` for it for more wider use cases.

Reply:
This is copied from https://github.com/rectorphp/rector-src/blob/f9916a10977a7000c126a01e17bf7c6f69ccf737/rules/PSR4/Rector/Namespace_/MultipleClassFileToPsr4ClassesRector.php#L174-L185, I guess it should be extracted somewhere.

Reply:
Did that.